### ISSUE TITLE
PackedScene not saving CollisionShape of StaticBody

### ISSUE THREAD

**Issue by TheSHEEEP** [godotengine/godot#28901]
**Godot version:**
3.1.1

**OS/device including version:**
Ubuntu 18.04, GeForce 1070

**Issue description:**
I use the following code to create a MeshInstance with a StaticBody and CollisionShape, then store it to a new scene:
```
# CreateMeshInstance
var meshInstance :MeshInstance = MeshInstance.new()
meshInstance.name = "CreatedMesh"
meshInstance.mesh = mesh
meshInstance.translation = Vector3(0.0, 0.0, 0.0)

# Create collision
var staticBody :StaticBody = StaticBody.new()
var collisionShape :CollisionShape = CollisionShape.new()
var shape :Shape = mesh.create_trimesh_shape()
collisionShape.shape = shape
collisionShape.owner = staticBody
staticBody.add_child(collisionShape)
meshInstance.add_child(staticBody)
staticBody.owner = meshInstance
				
# Store it in a scene
var packedScene : PackedScene = PackedScene.new()
packedScene.pack(meshInstance)
ResourceSaver.save("user://mesh.tscn", packedScene)
```
When adding the actual meshInstance to the scene, it works just well and I can see (via Debug view) the collision shape. The "mesh" variable is an existing mesh.

However, when I load and instance the stored "mesh.tscn", it lacks the CollisionShape and therefore won't have any functioning collision. Opening the "mesh.tscn" in the editor itself also shows this. There is only a MeshInstance with a StaticBody as child (which doesn't have a CollisionShape as child).

From what I gathered, you have to set the owner of everything (which is very cumbersome, btw. and shouldn't be necessary) when using PackedScene.pack(). And as the code shows, I do that.

**Steps to reproduce:**
See above.

**Project:**
[procmeshtest.zip](https://github.com/godotengine/godot/files/3182026/procmeshtest.zip)
Only tested in linux so far.
To execute the code, press space to save the mesh to its own file (and add the original MeshInstance to the scene). Press space again to load and instance the file.
Make sure to enable showing collision meshes or you won't see any difference.

**Comment by TheSHEEEP** [comment]
Now actually has a description, sorry for that. @akien-mga please remove it from the archive ;)

**Comment by TheSHEEEP** [comment]
.

**Comment by TheSHEEEP** [comment]
I added my project to the description. It's not totally minimal, but I think small enough. All the relevant code is one file.

**Comment by qarmin** [comment]
Duplicate of https://github.com/godotengine/godot/issues/28535 ?

**Comment by TheSHEEEP** [comment]
Quite possible that there is an underlying problem that affects not only StaticBody.

**Comment by santouits** [comment]
set the owner of the collision shape to the mesh instance

**Comment by TheSHEEEP** [comment]
Tried that. Makes no difference.

**Comment by santouits** [comment]
Yeah, you can't set the owner of a node to a node that isn't a parent, grandparent,etc of that node.
You do `collisionShape.owner = staticBody`, but staticBody isn't a parent of the collisionShape yet and it fails.
Delete `collisionShape.owner = staticBody` and add after` meshInstance.add_child(staticBody)` `collisionShape.owner = meshInstance`

**Comment by TheSHEEEP** [comment]
Okay, with those changes it finally works.
For completion's sake, here is how the above code was changed:
```
# Create collision
var staticBody :StaticBody = StaticBody.new()
var collisionShape :CollisionShape = CollisionShape.new()
var shape :Shape = mesh.create_trimesh_shape()
collisionShape.shape = shape
staticBody.add_child(collisionShape)
meshInstance.add_child(staticBody)
collisionShape.owner = meshInstance
staticBody.owner = meshInstance
```

I must say, though, I find this `owner` variable rather cumbersome. Why not just save everything under a node in PackedScene? If someone wanted certain children not included, they could always remove them beforehand. Or at least make the variable opt-out instead of opt-in (so that you have to set it if you **don't** want a node included).
The current behaviour is actually counter-intuitive.

Anyway, this is resolved now - except I suspect that I will hit #18293 before the end of my test ;)
